src/array.ts

```diff
@@ -118,3 +118,13 @@ export function makeStringCompareFn<TSortable>(
 export function localeCompareStrings(a: string, b: string): number {
   return a.localeCompare(b);
 }
+
+/**
+ * Select first element of an array if it contains exactly one element, else return null.
+ */
+export function firstIfSingle<T>(array: Maybe<Array<T>>): T | null {
```

Comment:
```suggestion
/** Returns the item from the given array if it contains exactly one item, otherwise null. */
export function soleItem<T>(array: Maybe<Array<T>>): T | null {
```